js/core/utils/data.js

```diff
@@ -62,12 +62,17 @@ var compileGetter = function(expr) {
         return function(obj, options) {
             options = prepareOptions(options);
             var functionAsIs = options.functionsAsIs,
+                defaultValueExists = "defaultValue" in options,
                 current = unwrap(obj, options);
 
             for(var i = 0; i < path.length; i++) {
-                if(!current) break;
+                if(!current) {
+                    break;
+                }
 
-                var next = unwrap(current[path[i]], options);
+                var useDefaultValue = defaultValueExists && typeUtils.isPlainObject(current) && !(path[i] in current),
```

Comment:
> typeUtils.isPlainObject(current)

What about cases with class instances?

```
class A {  };
assert.equal(GETTER("a")(new A(), { defaultValue: "default" }), "default");
```

Reply:
Does this answer @San4es' question?

https://github.com/DevExpress/DevExtreme/blob/cd50829772388821d91974517d382d5c58504b3f/testing/tests/DevExpress.core/utils.data.tests.js#L65-L77

Reply:
@AlekseyMartynov it's not like that. As far as I see, it is the test scenario for `path[i] in current` vs. `current.hasOwnProperty(path[i])`.

Reply:
Got it.
@mrjohn0011 changed to `isObject` in https://github.com/DevExpress/DevExtreme/pull/5212/commits/368010325748a32bebe6fa6e441dfe75ef7ed295

Reply:
Assert added:
https://github.com/DevExpress/DevExtreme/blob/4252cc738584de868d68f674e01866cf85b97d29/testing/tests/DevExpress.core/utils.data.tests.js#L86